samples/CommunityToolkit.Maui.Sample/MauiProgram.cs

```diff
@@ -193,7 +193,7 @@ static void RegisterEssentials(in IServiceCollection services)
 		services.AddSingleton<IDeviceInfo>(DeviceInfo.Current);
 		services.AddSingleton<IFileSaver>(FileSaver.Default);
 		services.AddSingleton<IFolderPicker>(FolderPicker.Default);
-		services.AddSingleton<IBadgeCounter>(BadgeCounter.BadgeCounter.Default);
+		services.AddSingleton<IBadge>(Badge.Default);
 		services.AddSingleton<ISpeechToText>(SpeechToText.Default);
 		services.AddSingleton<ITextToSpeech>(TextToSpeech.Default);
```

Comment:
Wondering if we can do better. What do you think to add the default implementation on our `UseCommunityToolkit`? If the user wants a different implementation, they can override it. Of course this change should be in another PR, but this come to my mind when I saw this change

Reply:
Pros of this approach:
- User writes less code and get everything out of the box
Cons of this approach:
- Create and register instances that are not used in my project at all
- it differs from .NET MAUI implementation that also doesn't register essentials (probably because it affects performance).

Reply:
> Create and register instances that are not used in my project at all

Aren't those lazy generated? So they will be mapped, but just instantiated when the user calls it for the first time?

> it differs from .NET MAUI implementation that also doesn't register essentials (probably because it affects performance).

Not sure why maui doesn't do that, maybe they didn't think about it?

Also, our lib isn't linker safe, so even if the dev doesn't use the code, it will be package by default

Reply:
Default creates instance immediately. We don't use Lazy, but probably should.
Do you know what should we do to make it linker friendly?

Reply:
I'll create a discussion then, since this is out of the scope of this PR.

> Do you know what should we do to make it linker friendly?

Yes, I need to find the reference... When Maui was in preview I tried to run the linker analyzer on the lib but didn't work for me, let's see if now the history will be different